plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/HiveFileIterator.java

```diff
@@ -111,12 +109,11 @@ private TrinoFileStatus getLocatedFileStatus(Iterator<TrinoFileStatus> iterator)
     }
 
     @VisibleForTesting
-    static boolean isHiddenFileOrDirectory(Path path)
+    static boolean isHiddenFileOrDirectory(Location location)
     {
         // Only looks for the last part of the path
-        String pathString = path.toUri().getPath();
-        int lastSeparator = pathString.lastIndexOf(SEPARATOR_CHAR);
-        return containsHiddenPathPartAfterIndex(pathString, lastSeparator + 1);
+        int lastSeparator = location.path().lastIndexOf('/');
```

Comment:
not sure it matters, but you might want to leave the assignment of location.path to a variable.  makes it clear the index is applied to the correct string... just a nit